### PR TITLE
fix: Add missing possibility for remove user group #10053 + delete address confirmation

### LINKED ISSUE #10053: Add missing possibility for remove user group + delete address confirmation

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #10053

### CODE REVIEW

File: feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.component.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { Subject, Subscription } from 'rxjs';
+import { filter, first, take } from 'rxjs/operators';
+import { ItemService } from '../../item.service';
+import { ConfirmationMessageComponent } from '../../message/confirmation/confirmation-message.component';
+import { ConfirmationMessageData } from '../../message/confirmation/confirmation-message.model';
+import { MessageService } from '../../message/services/message.service';
+import { BaseItem } from '../../organization.model';
+
+/**
+ * Reusable component in the my-company are to toggle the disabled state for
```
Comment (tobi-or-not-tobi) [inline feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.component.ts]: copy paste docs

File: feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  /**
+   * The key input can be used to add a custom key.
+   *
+   * Most _organization_ entities use the `code` key, but there is some variations.
+   */
+  @Input() key = 'code';
+  @Input() additionalParam?: string;
+
+  /**
```
Comment (tobi-or-not-tobi) [inline feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.component.ts:35]: outdated comments

File: feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.module.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { I18nModule, UrlModule } from '@spartacus/core';
+import { ConfirmationMessageModule } from '../../message/confirmation/confirmation-message.module';
+import { MessageModule } from '../../message/message.module';
+import { DeleteItemComponent } from './delete-item.component';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    RouterModule,
+    UrlModule,
```
Comment (tobi-or-not-tobi) [inline feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.module.ts]: `UrlModule` isn't used

File: feature-libs/organization/administration/styles/_card.scss [hunk 4]
```diff
@@ -10,7 +10,8 @@
     @extend %cardMain !optional;
     @extend %cardSection !optional;
 
-    cx-org-toggle-status {
+    cx-org-toggle-status,
```
Comment (tobi-or-not-tobi) [inline feature-libs/organization/administration/styles/_card.scss]: what about adding `host: { class: 'content-wrapper' }` to those instead?

**Review by tobi-or-not-tobi — DISMISSED** [review]
LGTM. Few nitpicks.

File: feature-libs/organization/administration/components/shared/item.service.ts [hunk 5]
```diff
@@ -60,6 +60,13 @@ export abstract class ItemService<T> {
     }
   }
 
+  /**
+   * Delete an item.
```
Comment (mszczepaniak) [inline feature-libs/organization/administration/components/shared/item.service.ts]:
```suggestion
   * Deletes an item.
```

File: feature-libs/organization/administration/assets/translations/en/units.i18n.ts [hunk 6]
```diff
@@ -187,5 +187,7 @@ export const unitAddress = {
   messages: {
     update:
       'Address {{ item.firstName }} {{ item.lastName }} updated successfully',
+    delete: 'Are you sure you want to delete this address?',
```
Comment (mszczepaniak) [inline feature-libs/organization/administration/assets/translations/en/units.i18n.ts]: We could also use the item here

File: feature-libs/organization/administration/components/user-group/services/user-group-item.service.ts [hunk 7]
```diff
@@ -46,4 +52,10 @@ export class UserGroupItemService extends ItemService<UserGroup> {
   protected getDetailsRoute(): string {
     return 'userGroupDetails';
   }
+
+  protected launchList() {
```
Comment (mszczepaniak) [inline feature-libs/organization/administration/components/user-group/services/user-group-item.service.ts:56]:
```suggestion
  protected goToList() {
```

File: feature-libs/organization/administration/styles/_card.scss [hunk 8]
```diff
@@ -9,10 +9,6 @@
     @extend %cardHeader !optional;
     @extend %cardMain !optional;
     @extend %cardSection !optional;
-
-    cx-org-toggle-status {
```
Comment (mszczepaniak) [inline feature-libs/organization/administration/styles/_card.scss:13]: anything in common with the delete functionality?

File: feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.module.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import { RouterModule } from '@angular/router';
+import { I18nModule } from '@spartacus/core';
+import { ConfirmationMessageModule } from '../../message/confirmation/confirmation-message.module';
+import { MessageModule } from '../../message/message.module';
+import { DeleteItemComponent } from './delete-item.component';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    RouterModule,
```
Comment (mszczepaniak) [inline feature-libs/organization/administration/components/shared/detail/delete-item-action/delete-item.module.ts]: I do not see an usage of the routing in the `delete-item.component.ts`

File: feature-libs/organization/administration/styles/_card.scss (hunk 8 above)
Comment (Pucek9) [reply feature-libs/organization/administration/styles/_card.scss:13]: https://github.com/SAP/spartacus/pull/10075#discussion_r534230659 - unified for consistency

File: feature-libs/organization/administration/components/user-group/services/user-group-item.service.ts (hunk 7 above)
Comment (Pucek9) [reply feature-libs/organization/administration/components/user-group/services/user-group-item.service.ts:56]: I used the same name as we have already in unit-address-item.service

File: feature-libs/organization/administration/assets/translations/en/units.i18n.ts (hunk 6 above)
Comment (Pucek9) [reply feature-libs/organization/administration/assets/translations/en/units.i18n.ts]: yes, but I'm not able to create better question (with context) in this case. Maybe some proposals? 😁

Comment (mszczepaniak) [reply feature-libs/organization/administration/assets/translations/en/units.i18n.ts]: 'Are you sure you want to delete {{ item.firstName }}?'

**Review by mszczepaniak — DISMISSED** [review]

**Review by mszczepaniak — APPROVED** [review]

**Pull request merged**